### DISCUSSION TITLE
Unable to query @refresh metadata

### DISCUSSION THREAD

**Discussion by ItsRobbAllen** [ravendb/ravendb#14962] [Q&A] [answered]
I am trying to set up a subscription that refreshes on a schedule (weekly, monthly, etc).

The subscription is 
`from "GeneratePaymentRequests" as req
where req.@metadata.@refresh == null`

What I expect is the refresh deletes the metadata, my generation runs, sets a new refresh tag for the next time it should run, and life goes on.

However, I absolutely cannot query @refresh. 

Currently, all the payment request documents do NOT have a refresh tag, but the above query returns 0 results.

If I switch it to 
`from "GeneratePaymentRequests" as req
where req.@metadata.@refresh != null`

It returns ALL the requests even though NONE of them have that metadata.

Am I doing something wrong here?

Edit: I am using 5.4.2 and this behavior happens in my local Raven instance (running in Docker) as well as all of our Cloud instances

**Comment by ayende** [comment, marked as answer]
You need to quote the fields:

```
from "GeneratePaymentRequests" as req where req.'@metadata'.'@refresh' == null
```

But your scenario is that you are checking the value of a field that isn't _there_, you need to use *exists()*:

```
from "GeneratePaymentRequests" as req where exists(req.'@metadata'.'@refresh')
```

**Reply by ItsRobbAllen** [reply]
The single quotes seem to work better, but I need to find documents that DO NOT have a refresh tag.

`from "GeneratePaymentRequests" as req where NOT exists(req.'@metadata'.'@refresh')` doesn't work, but `from "GeneratePaymentRequests" as req where true AND NOT exists(req.'@metadata'.'@refresh')` does :/

**Reply by ayende** [reply]
Yes, the later option is the right way to handle this

**Reply by loraderon** [reply]
@ayende I believe the `c.'@metadata'.'@refresh' == null` rather than `[true and] not exists(c.'@metadata'.'@refresh')` originates from https://www.ayende.com/blog/195393-A/ravendb-subscriptions-messaging-patterns. Perhaps consider updating that blog post and adding it to https://docs.ravendb.net/7.1/server/extensions/refresh?

**Reply by ayende** [reply]
https://issues.hibernatingrhinos.com/issue/RavenDB-25145/Ensure-c.metadata.refresh-null-works-for-subscriptions
